documentation/TechnicalDesignDocument.tex

```diff
@@ -209,8 +228,35 @@
     In this section build automation and its constituent components will be discussed.
     How we applied it, the software used and the targets and platforms supported are among a few.
 
+    The requirements for the project contain several important matters.
+    The project needs to be able to be developed and built from all OSes.
+    It needs to be able to switch hands between multiple developers easily, due to it being open source.
+    It should also obviously be thoroughly tested.
+    Because of this, the team, together with Chide decided that a continuous integration solution was necessary, and have been working on setting this up during the first part of the project.
+
+    The chosen CI system is Travis CI .
+    It is free for open source, and easy to use and learn.
+    After some research between multiple options the available documentation, resources, and functionality, such as the multi-os build options, made the team decide that Travis would be the best option.
+    There were a few problems with the setup, which will be detailed further below.
+    The current implementation of Travis CI:
+    \begin{itemize}
+        \item-Sets up a container for the project to run.
+        \item-Automatically builds the project
+        \item-Runs through all unit tests
+        \item-Allows live views and checks of the build.
+    \end{itemize}
+    This is currently always a Linux build, due to being unable to properly set up a Mac OS or Windows build.
+    This ensures that all commits have been properly tested.
+    Code review can be done without having to clone and test the code yourself.
```

Comment:
Does Travis review your code? Is this a feature we havn't used or is this a typo?